File: Android/src/androidMain/kotlin/com/taskbridge/android/repository/MessagesRepository.kt
package com.taskbridge.android.repository

import com.taskbridge.core.models.messages.AppMessage
import com.taskbridge.core.models.messages.AppMessageKey
import kotlinx.coroutines.flow.Flow

interface MessagesRepository {
    fun observeAll(): Flow<AppMessage>
    fun observe(types: List<AppMessageKey>): Flow<AppMessage>
    fun observe(type: AppMessageKey): Flow<AppMessage>
    suspend fun publish(message: AppMessage)
}

File: iOS/TaskBridge/Repository/MessagesRepository.swift
import Foundation
import TaskBridgeCore

@MainActor
protocol MessagesRepository {
    func observeAll() -> AsyncStream<AppMessage>
    func observe(types: [AppMessageKey]) -> AsyncStream<AppMessage>
    func observe(type: AppMessageKey) -> AsyncStream<AppMessage>
    func publish(message: AppMessage) async throws
}
